File: 은빈/자바/[PGS]같은숫자는싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        Stack<Integer> stack = new Stack<>();
        List<Integer> answer = new ArrayList<>();
        
        for(int val : arr) {
            if(!stack.contains(val)) {
                if(stack.size() != 0) {
                    answer.add(stack.pop());
                    stack.clear();
                }
                stack.push(val);
            }
        }
        
        if(stack.size() != 0) {
            answer.add(stack.pop());
        }

        return answer.stream().mapToInt(Integer::intValue).toArray();
    }
}

File: 은빈/파이썬/[PGS]같은숫자는싫어.py
def solution(arr):
    answer = []
    stack = []
    
    for num in arr:
        if num not in stack:
            if len(stack) != 0:
                answer.append(stack.pop())
            stack.clear()
            stack.append(num)
        
    if len(stack) != 0:
        answer.append(stack.pop())
    
    return answer